src/interfaces/chain.cpp

```diff
@@ -0,0 +1,453 @@
+template <typename T>
+class Ref : public std::reference_wrapper<T>
+{
+public:
+    using std::reference_wrapper<T>::reference_wrapper;
+    bool operator<(const Ref<T>& other) const { return this->get() < other.get(); }
+};
+
+class LockImpl : public Chain::Lock
+{
+    Optional<int> getHeight() override
+    {
+        int height = ::chainActive.Height();
+        if (height >= 0) {
+            return height;
+        }
+        return nullopt;
+    }
+    Optional<int> getBlockHeight(const uint256& hash) override
+    {
+        auto it = ::mapBlockIndex.find(hash);
+        if (it != ::mapBlockIndex.end() && it->second) {
+            if (::chainActive.Contains(it->second)) {
+                return it->second->nHeight;
+            }
+        }
+        return nullopt;
+    }
+    int getBlockDepth(const uint256& hash) override
+    {
+        const Optional<int> tip_height = getHeight();
+        const Optional<int> height = getBlockHeight(hash);
+        return tip_height && height ? *tip_height - *height + 1 : 0;
+    }
+    uint256 getBlockHash(int height) override { return ::chainActive[height]->GetBlockHash(); }
+    int64_t getBlockTime(int height) override { return ::chainActive[height]->GetBlockTime(); }
+    int64_t getBlockMedianTimePast(int height) override { return ::chainActive[height]->GetMedianTimePast(); }
+    bool haveBlockOnDisk(int height) override
+    {
+        CBlockIndex* block = ::chainActive[height];
+        return block && (block->nStatus & BLOCK_HAVE_DATA) && block->nTx > 0;
+    }
+    Optional<int> findFirstBlockWithTime(int64_t time) override
+    {
+        CBlockIndex* block = ::chainActive.FindEarliestAtLeast(time);
+        if (block) {
+            return block->nHeight;
+        }
+        return nullopt;
+    }
+    Optional<int> findFirstBlockWithTimeAndHeight(int64_t time, int height) override
+    {
+        for (CBlockIndex* block = ::chainActive[height]; block; block = ::chainActive.Next(block)) {
+            if (block->GetBlockTime() >= time) {
+                return block->nHeight;
+            }
+        }
+        return nullopt;
+    }
+    Optional<int> findPruned(int start_height, Optional<int> stop_height) override
+    {
+        if (::fPruneMode) {
+            CBlockIndex* block = stop_height ? ::chainActive[*stop_height] : ::chainActive.Tip();
+            while (block && block->nHeight >= start_height) {
+                if (!(block->nStatus & BLOCK_HAVE_DATA)) {
+                    return block->nHeight;
+                }
+                block = block->pprev;
+            }
+        }
+        return nullopt;
+    }
+    Optional<int> findFork(const uint256& hash, Optional<int>* height) override
+    {
+        const CBlockIndex *block{nullptr}, *fork{nullptr};
+        auto it = ::mapBlockIndex.find(hash);
+        if (it != ::mapBlockIndex.end()) {
+            block = it->second;
+            fork = ::chainActive.FindFork(block);
+        }
+        if (height) {
```

Comment:
```
2018-09-19 14:39:06 clang-tidy(pr=10973): src/interfaces/chain.cpp:121:13: warning: implicit conversion 'Optional<int> *' (aka 'boost::optional<int> *') -> bool [readability-implicit-bool-conversion]
```

Reply:
@practicalswift You know, the signal/noise ratio of these comments would be higher if you omitted the timestamp (which isn't interesting) and the file path (which is redundant, because that's where the comment is attached). Currently in the web UI, you always have to scroll each comment horizontally to see what it's actually about.

status quo:
<img width="676" alt="bildschirmfoto 2018-09-23 um 18 40 53" src="https://user-images.githubusercontent.com/425166/45930419-5f36e980-bf60-11e8-866f-102591e58d38.png">

proposed:
<img width="664" alt="bildschirmfoto 2018-09-23 um 18 41 21" src="https://user-images.githubusercontent.com/425166/45930424-6958e800-bf60-11e8-841b-1e553ca9d7b6.png">